cmd/k8s-deployment-scaler/main.go

```diff
@@ -0,0 +1,354 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Define the global clientset variable using kubernetes.Interface
+var clientset kubernetes.Interface
+
+// DeploymentInfo stores the relevant information about a Deployment
+type DeploymentInfo struct {
+	Replicas   int32
```

Comment:
Are replicas not a part of the deployment already? If so, doesn't this make this struct unnecessary

Reply:
YAGNI. Thank you.